en/asgardeo/docs/quick-starts/nextjs.md

```diff
@@ -0,0 +1,312 @@
+
+    cd asgardeo-nextjs
+
+    yran install
+
+    yarn dev
+    ```
+
+=== "pnpm"
+
+    ``` bash
+    pnpm create next-app --typescript asgardeo-nextjs
+
+    cd asgardeo-nextjs
+
+    pnpm install
+
+    pnpm run dev
+    ```
+
+## Install Auth.js library
+
+Auth.js is a lightweight JavaScript library designed for simplifying authentication workflows in web application. The [Asgardeo provider for Auth.js](https://authjs.dev/reference/core/providers/asgardeo){:target="_blank"}  offers all the components and hooks you need to integrate your app with {{product_name}}.To get started, simply add Auth.js library to the project. Make sure to stop the dev server started in the previous step.
+
+=== "npm"
+
+    ``` bash
+    npm install next-auth@beta
+    ```
+
+=== "yarn"
+
+    ``` bash
+    yarn add next-auth@beta
+    ```
+
+=== "pnpm"
+
+    ``` bash
+    pnpm add next-auth@beta
+    ```
+
+Next, generate **`AUTH_SECRET`** environment variable. 
+
+=== "npm"
+
+    ``` bash
+    npx auth secret
+
+    ```
+=== "yarn"
+
+    ``` bash
+    yarn dlx auth secret
+    ```
+
+=== "pnpm"
+
+    ``` bash
+    pnpm dlx auth secret
+    ```
+
+Add following entries to the  .env or .env.local file, and make sure to replace the placeholders in the following code with the **`client-id`**, **`client-secret`** and **`issuer`** values you copied in **Step-1** during the application registration in the {{product_name}} console. 
+
+
+```bash title=".env.local"
+    AUTH_ASGARDEO_ID="<your-app-client-id>"
+    AUTH_ASGARDEO_SECRET="<your-app-client-secret>"
+    AUTH_ASGARDEO_ISSUER="<your-app-issuer-url>"
+
+```
+
+
+
+## Create the `auth.js` configuration File
+
+Create a file called `/src/auth.ts'`. 
+
+```bash
+
+touch /src/auth.ts
+
+```
+
+Add {{product_name}} as a provider in the `/src/auth.ts'` file.
+
+```javascript title="auth.ts"
+import NextAuth from "next-auth"
+import Asgardeo from "next-auth/providers/asgardeo"
+
+export const { handlers, signIn, signOut, auth } = NextAuth({
+  providers: [Asgardeo({
+    issuer: process.env.AUTH_ASGARDEO_ISSUER
+  })],
+})
+
+```
+
+
+
+Create a Route Handler file in the `src/app/api/auth/[...nextauth]/route.ts` location. 
+
+```bash
+mkdir -p src/app/api/auth/\[...nextauth\]
+
+touch mkdir -p src/app/api/auth/\[...nextauth\]/route.ts
+
+```
+
+!!! Note
+    The directory `src/app/api/auth/[...nextauth]/route.ts` in a Next.js project is used to define a dynamic API route for handling authentication. The `[...nextauth]` is a catch-all route that processes multiple authentication-related requests such as sign-in, sign-out, and session management. The route.ts file specifies the logic for these operations, typically by exporting handlers for HTTP methods like GET and POST. This setup centralizes authentication logic, supports OAuth providers like Google or GitHub, and integrates seamlessly into Next.js applications for secure and scalable authentication workflows.
```

Comment:
The directory `src/app/api/auth/[...nextauth]/route.ts` -> The file `src/app/api/auth/[...nextauth]/route.ts``